src/targets/gpu/include/migraphx/gpu/context.hpp

```diff
@@ -272,8 +274,40 @@ struct context
 
         auto v_streams        = v.at("streams");
         std::size_t n_streams = v_streams.without_key().to<std::size_t>();
+        this->current_device  = std::make_shared<hip_device>(0, n_streams);
+#ifdef MIGRAPHX_HAS_FIND_2_API
+        {
+            auto v_gfx_arch = v.at("gfx_arch").to<std::string>();
+
+            auto v_cu_count        = v.at("cu_count");
+            std::size_t n_cu_count = v_cu_count.without_key().to<std::size_t>();
 
-        this->current_device = std::make_shared<hip_device>(0, n_streams);
+            std::string v_miopen_version = v.at("miopen_version").to<std::string>();
+
+            std::string current_gfx_arch       = this->current_device->get_device_name();
+            std::size_t current_cu_count       = this->current_device->get_cu_count();
+            std::string current_miopen_version = get_miopen_version();
+            if(n_cu_count != current_cu_count or v_gfx_arch != current_gfx_arch)
```

Comment:
If the gfx arch is different then we need to throw an error, not just print a message, but only if we have binary objects we compiled or saved from miopen.(lets assume miopen's saved solution have saved binaries). However we would need to throw this error in something like finalize.

Reply:
Error would prevent users to run the model altogether.  We want to enable transferring or  sharing  the model file (.mxr or YModel) across different platforms.

Reply:
> Warnings would warn them about gfx arch mismatch but still allow them to run the model

Ah because we can load gfx900 on MI100 machines. But if the model is built for gfx10, or its built for gfx906 and is trying to load on vega10(gfx900), it should be an hard error.

Either way, there should only be a message when we have code objects in the model. For example, there shouldn't be any message if everything is rocblas calls.

Reply:
> Either way, there should only be a message when we have code objects in the model 

I see you point now. HIP code objects would be specific to GFX Arch.   I can check for `code_object ` in that case.

Reply:
>  I can check for code_object in that case. 

I don't see any practical model without that, so I think I should just error out.